flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/AbstractStreamOperator.java

```diff
@@ -160,7 +160,7 @@ public void setup(StreamTask<?, ?> containingTask, StreamConfig config, Output<S
 			if (config.isChainStart()) {
 				operatorMetricGroup.getIOMetricGroup().reuseInputMetricsForTask();
 			}
-			if (config.isChainEnd()) {
+			if (!config.getNonChainedOutputs(getUserCodeClassloader()).isEmpty()) {
```

Comment:
You've missed the same change but for `AbstractStreamOperatorV2` (I would expect `MultipleInputStreamTaskTest#testOperatorMetricReuse` would cover for that)